src/Common/Abstracts/Custom_Table_Repository.php

```diff
@@ -0,0 +1,1081 @@
+<?php
+/**
+ * The custom table repository.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Abstracts
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Common\Abstracts;
+
+use TEC\Common\Contracts\Custom_Table_Repository_Interface as Repository_Interface;
+use TEC\Common\StellarWP\SchemaModels\Contracts\SchemaModel as Model;
+use TEC\Common\StellarWP\Schema\Tables\Contracts\Table as Table_Interface;
+use TEC\Common\StellarWP\SchemaModels\Relationships\ManyToManyWithPosts;
+use TEC\Common\StellarWP\DB\DB;
+use Tribe__Promise as Promise;
+use RuntimeException;
+use Generator;
```

Comment:
2 small things:

1. Alphabetize the imports. This applies to all files.
2. Include `WP_Post` as an import. Not strictly needed for the code, but `WP_Post` _is_ referenced in docblocks.

```suggestion
use Generator;
use WP_Post;
```